extension-manifest-v3/src/background/adblocker.js

```diff
@@ -223,15 +235,23 @@ async function executeScriptlets(tabId, scripts) {
     debugMarker = () => '';
   }
 
+  const nonceSize = 20;
+  const nonce = createRandomString(nonceSize);
   // the scriptlet code that contains patches for the website
-  const codeRunningInPage = `(function(){
+  const codeRunningInPage = `(function(){const nonce = '${nonce}';
 ${debugMarker('run scriptlets (executing in "page world")')}
 ${scripts.join('\n\n')}}
 )()`;
 
   // wrapper to break the "isolated world" so that the patching operates
   // on the website, not on the content script's isolated environment.
-  function codeRunningInContentScript(code) {
+  function codeRunningInContentScript(code, nonce, nonceSize) {
+    if (window.trustedTypes && window.trustedTypes.createPolicy) {
+      window.trustedTypes.createPolicy('default', {
```

Comment:
I think you must use returned policy to create a script, but here the returned policy is not set to any variable.

Look here https://developer.mozilla.org/en-US/docs/Web/API/TrustedScript

Reply:
Ok, you made a fix in the same time I wrote this.